src/silvimetric/cli/cli.py

```diff
@@ -83,23 +96,27 @@ def initialize_cmd(app: ApplicationConfig, bounds: Bounds, crs: pyproj.CRS,
 @cli.command('shatter')
 @click.argument("pointcloud", type=str)
 @click.option("--workers", type=int, default=12)
-@click.option("--threads", default=4, type=int)
+@click.option("--threads", type=int, default=4)
 @click.option("--bounds", type=BoundsParamType(), default=None)
 @click.option("--tilesize", type=int, default=None)
 @click.option("--watch", is_flag=True, default=False, type=bool)
 @click.option("--report", is_flag=True, default=False, type=bool)
+@click.option("--date", type=click.DateTime(['%Y-%m-%d','%Y-%m-%dT%H:%M:%SZ']), required=True, multiple=True)
 @click.option("--dasktype", default='cluster', type=click.Choice(['cluster',
               'threads', 'processes', 'single-threaded']))
 @click.pass_obj
 def shatter_cmd(app, pointcloud, workers, bounds, threads, watch, report,
-                dasktype, tilesize):
+                dasktype, tilesize, date):
+    if len(date) > 2:
+        raise Exception('"--date" must be either 1 or 2 values.')
```

Comment:
Click has validation functions (see BoundsTypeParam, etc). We should use those instead of checking dates like this.